src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -165,6 +165,99 @@ function scrollToActive() {
   });
 }
 
+/*******************************************************************************
+ * Search
+ */
+var changeShortcutText = () => {
+  // Change the search hint to `meta key` if we are a Mac
+  let forms = document.querySelectorAll("form.bd-search");
+  var isMac = window.navigator.platform.toUpperCase().indexOf("MAC") >= 0;
+  if (isMac) {
+    forms.forEach(
+      (f) => (f.querySelector("kbd.kbd-shortcut__modifier").innerText = "⌘")
+    );
+  }
+};
+
+var findSearchInput = () => {
+  // find the search form(s) on the page
+  let forms = document.querySelectorAll("form.bd-search");
+  if (!forms.length) {
+    // no search form found
+    return;
+  } else {
+    var form;
+    if (forms.length == 1) {
+      // there is exactly one search form (persistent or hidden)
+      form = forms[0];
+    } else {
+      // must be at least one persistent form, use the first persistent one
+      form = document.querySelector(
+        "div:not(.search-button__search-container) > form.bd-search"
+      );
+    }
+    return form.querySelector("input");
+  }
+};
+
+var toggleSearchField = () => {
+  // focus/unfocus the search field (and if it's an auto-hiding one,
+  // show/hide it too)
+  let input = findSearchInput();
+  let button = document.getElementById("bd-search-button");
+  // if the input field is the hidden one (the one associated with the
+  // search button) then toggle the button state (to show/hide the field)
+  if (
+    input.parentElement.parentElement.classList.contains(
```

Comment:
how about rather than hard-coding the two `parentElements` we do something like:

(messy pseudo code)

```javascript
if document.querySelector(".search-button__search-container input") == input
```

I think that might be a bit less-brittle and more obvious what we mean

Reply:
done in 39a4c69